src/metabase/models/params/chain_filter.clj

```diff
@@ -370,14 +375,16 @@
 
 (mu/defn ^:private chain-filter-mbql-query
   "Generate the MBQL query powering `chain-filter`."
-  [field-id                          :- ms/PositiveInt
+  [field                             :- Constraint
```

Comment:
I'm not very happy about calling it `field`, but using `constraint` feels like it would be even more confusing, and I have no better ideas for the name. `param` is also loaded word...